gateway/src/node/lp_listener/handler.rs

```diff
@@ -104,36 +104,14 @@ impl LpConnectionHandler {
         // Track total LP connections handled
         inc!("lp_connections_total");
 
-        // For LP, we need:
-        // 1. Gateway's keypair (from local_identity)
-        // 2. Client's public key (will be received during handshake)
-        // 3. PSK (pre-shared key) - for now use a placeholder
-
-        // Derive LP keypair from gateway's ed25519 identity using proper conversion
-        // This creates a valid x25519 keypair for ECDH operations in Noise protocol
-        let x25519_private = self.state.local_identity.private_key().to_x25519();
-        let x25519_public = self
-            .state
-            .local_identity
-            .public_key()
-            .to_x25519()
-            .map_err(|e| {
-                GatewayError::LpHandshakeError(format!(
-                    "Failed to convert ed25519 public key to x25519: {}",
-                    e
-                ))
-            })?;
-
-        let lp_private = LpPrivateKey::from_bytes(x25519_private.as_bytes());
-        let lp_public = PublicKey::from_bytes(x25519_public.as_bytes()).map_err(|e| {
-            GatewayError::LpHandshakeError(format!("Failed to create LP public key: {}", e))
-        })?;
-
-        let gateway_keypair = Keypair::from_keys(lp_private, lp_public);
+        // AIDEV-NOTE: Gateway LP handshake with Ed25519-only API
+        // The state machine now accepts only Ed25519 keys and internally derives X25519 keys.
+        // This simplifies the API by removing manual key conversion from the caller.
+        // Gateway's Ed25519 identity is used for both PSQ authentication and X25519 derivation.
 
         // Receive client's public key and salt via ClientHello message
         // The client initiates by sending ClientHello as first packet
-        let (client_pubkey, salt) = match self.receive_client_hello().await {
+        let (client_pubkey, client_ed25519_pubkey, salt) = match self.receive_client_hello().await {
```

Comment:
client's pubkey (i.e. the x25519) seems to be unused? is this expected?

Reply:
ah. right. we derive it from ed25519 - in that case, do we still need to include them in the ClientHello?